core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -153,6 +185,45 @@ private[spark] class DiskBlockManager(conf: SparkConf, deleteFilesOnStop: Boolea
     }
   }
 
+  /**
+   * Get the list of configured local dirs storing merged shuffle blocks created by executors
+   * if push based shuffle is enabled. Note that the files in this directory will be created
+   * by the external shuffle services. We only create the merge_manager directories and
+   * subdirectories here because currently the shuffle service doesn't have permission to
+   * create directories under application local directories.
+   */
+  private def createLocalDirsForMergedShuffleBlocks(conf: SparkConf): Unit = {
+    if (Utils.isPushBasedShuffleEnabled(conf)) {
+      // Will create the merge_manager directory only if it doesn't exist under any local dir.
+      Utils.getConfiguredLocalDirs(conf).foreach { rootDir =>
+        try {
+          val mergeDir = new File(rootDir, MERGE_MANAGER_DIR)
+          // This executor does not find merge_manager directory, it will start creating them.
+          // It's possible that the other executors launched at the same time may also reach here
+          // but we are working on the assumption that the executors launched around the same time
+          // will have the same set of application local directories.
```

Comment:
This comment needs to be updated as well. We are no longer working with this assumption.
`but we are working on the assumption that the executors launched around the same time
          // will have the same set of application local directories.`
 Also move it inside the `if` block below.